crates/sui-framework/sources/crypto/elliptic_curve.move

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Library for Elliptic Curve operations on chain. We specifically support the Ristretto-255 sub-group.
+module sui::elliptic_curve {
+    use std::vector;
+
+    ///////////////////////////////////
+    /// Elliptic Curve structs
+    ///////////////////////////////////
+
+    /// Represents a point on the Ristretto-255 subgroup.
+    struct RistrettoPoint has copy, drop, store {
+        // A 32-byte representation of the group element.
+        value: vector<u8>
+    }
+
+    /// Represents a scalar within the Curve25519 prime-order group.
+    struct Scalar has copy, drop, store {
+        // A 32-byte representation of the scalar
+        value: vector<u8>
+    }
+
+    ///////////////////////////////////
+    /// Private 
+    ///////////////////////////////////
+
+    /// @param value: The value to commit to
+    /// @param blinding_factor: A random number used to ensure that the commitment is hiding.
+    native fun native_create_pedersen_commitment(value: vector<u8>, blinding_factor: vector<u8>): vector<u8>;
+
+    /// @param self: bytes representation of an EC point on the Ristretto-255 subgroup 
+    /// @param other: bytes representation of an EC point on the Ristretto-255 subgroup 
+    /// A native move wrapper around the addition of Ristretto points. Returns self + other.
+    native fun native_add_ristretto_point(point1: vector<u8>, point2: vector<u8>): vector<u8>;
+
+    /// @param self: bytes representation of an EC point on the Ristretto-255 subgroup
+    /// @param other: bytes representation of an EC point on the Ristretto-255 subgroup 
+    /// A native move wrapper around the subtraction of Ristretto points. Returns self - other.
+    native fun native_subtract_ristretto_point(point1: vector<u8>, point2: vector<u8>): vector<u8>;
+
+    /// @param value: the value of the to-be-created scalar
+    /// TODO: Transfer this into a Move function some time in the future.
+    /// A native move wrapper for the creation of Scalars on Curve25519.
+    native fun native_scalar_from_u64(value: u64): vector<u8>;
+
+
+    /// @param value: the bytes representation of the scalar.
+    /// TODO: Transfer this into a Move function some time in the future.
+    /// A native move wrapper for the creation of Scalars on Curve25519.
+    native fun native_scalar_from_bytes(bytes: vector<u8>): vector<u8>;
+
+    ///////////////////////////////////
+    /// Public
+    ///////////////////////////////////
+    
+    // Scalar
+    ///////////////////////
+
+    /// Create a field element from u64
+    public fun new_scalar_from_u64(value: u64): Scalar {
+        Scalar {
+            value: native_scalar_from_u64(value)
+        }
+    }
+
+    /// Create a pedersen commitment from two field elements
+    public fun create_pedersen_commitment(value: Scalar, blinding_factor: Scalar): RistrettoPoint {
```

Comment:
Although it's not clear where we need that (because we reveal a blinding factor and in most privacy preserving apps we don't open commitments publicly), I'll approve for now as it might be useful in the future for fraud proofs etc.

Reply:
I'll keep this for this PR, but if I can implement the example without this, I'll remove it with the PR.